test/foundry/dapp/Application.t.sol

```diff
@@ -431,14 +452,22 @@ contract ApplicationTest is ERC165Test {
         bytes memory output = _getOutput(name);
         OutputValidityProof memory proof = _getProof(name);
 
+        _testERC20Fail(output, proof);
+        _testERC20Success(output, proof);
```

Comment:
I don't think we need to separate these into two distinct functions.
We could have just `_testERC20Transfer`, which runs the failure case and then the success case.

Reply:
It has to be split into 2. Because in the test case `testERC20TransferToENS`, we need to call `_mockENS` between the 2. Because at the end of `_testERC20Fail`, it called `clearMockedCalls`

Reply:
Although we could call `_mockENS` at the end of `_testERC20Fail` after `clearMockedCalls`, it's not the proper place for `_mockENS` to be in. So I don't mind splitting into 2.

Reply:
Yeah we could do that too